doc/reference/reference_lua/json.rst

```diff
@@ -116,6 +119,180 @@ Below is a list of all ``json`` functions and members.
     :ref:`Sum a JSON field for all tuples <c_lua_tutorial-sum_a_json_field>`
     to see how ``json.decode()`` can fit in an application.
 
+.. _json-serialize:
+
+**__serialize parameter:**
+
+The JSON output structure can be specified with ``__serialize``:
+
+* 'seq', 'sequence', 'array' - table encoded as an array
+* 'map', 'mapping' - table encoded as a map
+* function - the meta-method called to unpack serializable representation
+  of table, cdata or userdata objects
+
+Serializing 'A' and 'B' with different ``__serialize`` values brings different
+results:
+
+.. code-block:: tarantoolsession
+
+    tarantool> json.encode(setmetatable({'A', 'B'}, { __serialize="seq"}))
+    ---
+    - '["A","B"]'
+    ...
+    tarantool> json.encode(setmetatable({'A', 'B'}, { __serialize="map"}))
+    ---
+    - '{"1":"A","2":"B"}'
+    ...
+    tarantool> json.encode({setmetatable({f1 = 'A', f2 = 'B'}, { __serialize="map"})})
+    ---
+    - '[{"f2":"B","f1":"A"}]'
+    ...
+    tarantool> json.encode({setmetatable({f1 = 'A', f2 = 'B'}, { __serialize="seq"})})
+    ---
+    - '[[]]'
+    ...
+
+.. _json-module_cfg:
+
+.. function:: cfg(table)
+
+    Set values that affect the behavior of :ref:`json.encode <json-encode>`
+    and :ref:`json.decode <json-decode>`.
+
+    The values are all either integers or boolean ``true``/``false``.
+
+    .. container:: table
+
+        .. rst-class:: left-align-column-1
+        .. rst-class:: center-align-column-2
+        .. rst-class:: left-align-column-3
+
+        +---------------------------------+---------+-------------------------------------------+
+        | Option                          | Default | Use                                       |
+        +=================================+=========+===========================================+
+        | ``cfg.encode_max_depth``        |   128   | Max recursion depth for encoding          |
+        +---------------------------------+---------+-------------------------------------------+
+        | ``cfg.encode_deep_as_nil``      |  false  | A flag saying whether to crop tables       |
+        |                                 |         | with nesting level deeper than            | 
+        |                                 |         | ``max_depth``.                            |
```

Comment:
To avoid confusion, please specify the full name:

``max_depth`` --> ``cfg.encode_max_depth`